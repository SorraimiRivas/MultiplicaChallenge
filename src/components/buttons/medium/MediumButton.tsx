import {Text, TouchableOpacity} from 'react-native';
import React, {FC} from 'react';
import styles from './styles';

type Props = {
  title: string;
  onPress: () => void;
};

const MediumButton: FC<Props> = ({title, onPress}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

export default MediumButton;
