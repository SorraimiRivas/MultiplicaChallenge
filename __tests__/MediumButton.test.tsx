import React from 'react';
import renderer from 'react-test-renderer';
import MediumButton from '../src/components/buttons/medium/MediumButton';

test('renders MediumButton correctly', () => {
  const component = renderer.create(
    <MediumButton title="Test" onPress={() => null} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
