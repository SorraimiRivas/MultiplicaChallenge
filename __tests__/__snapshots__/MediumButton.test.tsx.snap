// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MediumButton correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "height": 50,
      "marginTop": 40,
      "opacity": 1,
      "padding": 14,
      "width": 170,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Avenir",
        "fontSize": 12,
        "fontWeight": "800",
      }
    }
  >
    Test
  </Text>
</View>
`;
